### PR TITLE
Fix DDC compilation error for the Dart Debug Extension

### PR DESCRIPTION
**Pull request by elliette**
Fixes https://github.com/dart-lang/webdev/issues/1506

Explicitly casts our tab list (which the compiler thought was `List<dynamic>`) to `List<Tab>`.

This is a known limitation of the Dart JS package (https://github.com/dart-lang/sdk/tree/master/pkg/js#list-instances-coming-from-javascript-will-always-be-listdynamic).

### CODE REVIEW

File: dwds/debug_extension/web/background.dart [hunk 1]
```diff
@@ -177,9 +177,10 @@ void _startDebugging(DebuggerTrigger debuggerTrigger) {
   // Extracts the extension backend port from the injected JS.
   var attachDebuggerToTab = allowInterop(_attachDebuggerToTab);
 
-  queryTabs(getCurrentTabQuery, allowInterop((List<Tab> tabs) {
-    if (tabs != null && tabs.isNotEmpty) {
-      attachDebuggerToTab(tabs[0]);
+  queryTabs(getCurrentTabQuery, allowInterop((List tabs) {
+    final tabList = List<Tab>.from(tabs);
+    if (tabList != null && tabList.isNotEmpty) {
```
Comment (natebosch) [inline dwds/debug_extension/web/background.dart:182]: `tabList` will never be null here

File: dwds/debug_extension/web/background.dart [hunk 2]
```diff
@@ -177,9 +177,10 @@ void _startDebugging(DebuggerTrigger debuggerTrigger) {
   // Extracts the extension backend port from the injected JS.
   var attachDebuggerToTab = allowInterop(_attachDebuggerToTab);
 
-  queryTabs(getCurrentTabQuery, allowInterop((List<Tab> tabs) {
-    if (tabs != null && tabs.isNotEmpty) {
-      attachDebuggerToTab(tabs[0]);
+  queryTabs(getCurrentTabQuery, allowInterop((List tabs) {
+    final tabList = List<Tab>.from(tabs);
```
Comment (natebosch) [inline dwds/debug_extension/web/background.dart:181]: we only ever us the first tab, do we need to create the full list for multiple tabs?

```dart
  if (tabs.isNotEmpty) {
    attachDebuggerToTab(tabList.first as Tab);
  }
```

File: dwds/debug_extension/web/background.dart (hunk 1 above)
Comment (elliette) [reply dwds/debug_extension/web/background.dart]: Ah good point!

File: dwds/debug_extension/web/background.dart (hunk 2 above)
Comment (elliette) [reply dwds/debug_extension/web/background.dart]: Good idea, done!

Also discovered a few other type errors when compiled with DDC, so I've fixed those as well.

File: dwds/debug_extension/web/background.dart [hunk 3]
```diff
@@ -363,7 +364,8 @@ void _forwardMessageToExternalExtensions(
 }
 
 void _notifyPanelScriptOfChanges(List<DevToolsPanel> panels) {
```
Comment (natebosch) [inline dwds/debug_extension/web/background.dart:366]:
```suggestion
void _notifyPanelScriptOfChanges(List panels) {
```

**Review by natebosch — APPROVED** [review]

**Pull request merged**